torchvision/csrc/io/decoder/gpu/demuxer.h

```diff
@@ -0,0 +1,241 @@
+extern "C" {
+#include <libavcodec/avcodec.h>
+#include <libavcodec/bsf.h>
+#include <libavformat/avformat.h>
+#include <libavformat/avio.h>
+}
+
+class Demuxer {
+ private:
+  AVFormatContext* fmtCtx = NULL;
+  AVBSFContext* bsfCtx = NULL;
+  AVPacket pkt, pktFiltered;
+  AVCodecID eVideoCodec;
+  uint8_t* dataWithHeader = NULL;
+  bool bMp4H264, bMp4HEVC, bMp4MPEG4;
+  unsigned int frameCount = 0;
+  int iVideoStream;
+  int64_t userTimeScale = 0;
+  double timeBase = 0.0;
+
+ public:
+  Demuxer(const char* filePath, int64_t timeScale = 1000 /*Hz*/) {
```

Comment:
For the (near) future: we will need to add support from reading from memory.

Also, is this `timeScale` used anywhere? Or do we plan to use it anywhere?

Reply:
https://github.com/pytorch/vision/issues/5148